### PR TITLE
[0.10.0] Enrollment List Printout

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
## Features
- Enrollment List Printout
  Resolves FRO-115 [(link)](https://linear.app/skiso/issue/FRO-115/enrollment-list-printout)
  - Enrollment Printout Header
    Resolves FRO-118

## Code Changes
- Generalized Student List Printout
  Resolves FRO-117
- `sortStudents` helper function

## Fixes
- Person Card in Classroom Student List only shows nickname
- Moved Electives pages URL to match convention
  - `/learn/elective` → `/learn/electives`
  - `/teach/elective` → `/teach/electives`

## Preview
This is the current main feature PR. Preview available at https://pr.mysk.school/.

### CODE REVIEW

File: components/classes/ClassStudentList.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ const ClassStudentList: StylableFC<{
         // Download the file
         window.location.href = URL.createObjectURL(mergedVCard);
 
-        va.track("Save Class VCards", { number: `M.${classNumber}` });
+        va.track("Save Class VCards", { number: `M.${classroom.number}` });
```
Comment (CursorWind) [inline components/classes/ClassStudentList.tsx:63]: Restructuring code this way makes it much easier to understand and use in the future.

File: components/classes/ClassroomPrintoutHeader.tsx
Comment (CursorWind) [inline components/classes/ClassroomPrintoutHeader.tsx:1]: Why would you separate the classroom and elective printout? It has the same style and elements? Is there a specific reason for that?

**Review by CursorWind — APPROVED** [review]
LGTM

File: components/classes/ClassroomPrintoutHeader.tsx
Comment (SiravitPhokeed) [reply components/classes/ClassroomPrintoutHeader.tsx:1]: There are quite a few differences as noted in your own meeting notes. For example, Classroom Printout shows the Classroom on the header, while Enrollment Printout shows the name of the Elective Subject. Plus, this allows for more flexibility when reusing Student List Printout in other contexts in the future.

**Review by smartwhatt — APPROVED** [review]
LGTM!!

File: components/elective/ElectiveStudentListCard.tsx [hunk 2]
```diff
@@ -47,34 +51,24 @@ const ElectiveStudentListCard: StylableFC<{
...
-            : null,
-        })),
-      // Sort by number in classroom, none at top.
-      (student) => student.class_no || 0,
-    ),
-    // Sort by classroom number, none at bottom. Students must have a Classroom
-    // to enroll in an Elective Subject anyway so this is just a way to hide
-    // if things go wrong but not being dishonest, according to @smartwhatt.
-    (student) => student.classroom?.number || 1000,
+  const students = sortStudents(
```
Comment (Sun-52) [inline components/elective/ElectiveStudentListCard.tsx:54]: Turning parts of the sorting process into a function is good for making the code easier to understand and develop in the future.

**Review by Sun-52 — CHANGES_REQUESTED** [review]
Print
I clicked print from the ElectiveStudentListCard and I got this.

**Comment by SiravitPhokeed** [comment]
> I clicked print from the ElectiveStudentListCard and I got this.

@Sun-52 Sorry about that. It looks like there was a problem with the domain and the preview deployment didn’t deploy correctly.
No code changes were made, but have a look at the preview again.

**Review by Sun-52 — APPROVED** [review]
LGTM

**Pull request merged**